Api/V8/Service/ModuleService.php

```diff
@@ -111,23 +115,83 @@ public function getRecords(GetModulesParams $params, Request $request)
         // negative numbers are validated in params
         $offset = $number !== 0 ? ($number - 1) * $size : $number;
         $realRowCount = $this->beanManager->countRecords($module, $where);
-        $limit = $size === BeanManager::DEFAULT_ALL_RECORDS ? BeanManager::DEFAULT_LIMIT : $size;
+        $limit = $size === BeanManager::DEFAULT_ALL_RECORDS
+            ? BeanManager::DEFAULT_LIMIT : $size;
         $deleted = $params->getDeleted();
 
         if (empty($fields)) {
             $fields = $this->beanManager->getDefaultFields($bean);
         }
 
-        $beanListResponse = $this->beanManager->getList($module)
-            ->orderBy($orderBy)
-            ->where($where)
-            ->offset($offset)
-            ->limit($limit)
-            ->max($size)
-            ->deleted($deleted)
-            ->fields($this->beanManager->filterAcceptanceFields($bean, $fields))
-            ->fetch();
+        // Detect if bean has email field
+        if ((property_exists($bean, 'email1') && strpos($where, 'email1') !== false)
+            || (property_exists($bean, 'email2') && strpos($where, 'email2') !== false)
+        ) {
+            $selectedFields = strtolower($module) . '.' . implode(',' . strtolower($module[0]) . '.', $fields);
+            $selectedModule = strtolower($module);
+
+            // Selects Module or COUNT(*) and will add one to the query.
+            $idSelect = 'SELECT ' . $selectedModule . '.id ';
+            $countSelect = 'SELECT COUNT(*) AS cnt ';
+
+            $quotedCountSelect = $db->quote($countSelect);
+
+            // Email where clause
+            $fromQuery
+                = 'FROM email_addresses JOIN email_addr_bean_rel ON email_addresses.id = email_addr_bean_rel.email_address_id JOIN '
+                . $selectedModule . ' ON ' . $selectedModule . '.id = email_addr_bean_rel.bean_id ';
+            $modifiedWhere = str_replace('accounts.email1',
```

Comment:
Hey, I still got filter error that responses email1 not found at version 7.11.15.

And I find out this line

```
$modifiedWhere = str_replace('accounts.email1',
```

should be replaced to 

```
$modifiedWhere = str_replace($selectedModule.'.email1',
```
Because not only the accounts has the email1 and email2

Reply:
True! I was trying to get email for Leads module, but this was driving me crazy until I modified `Api/V8/Service/ModuleService.php` to replace accounts with `$selectedModule`.

Reply:
This ticket must be re-opened. You still cannot search Leads by email1.
Thanks @owl-mark for your fix.